autograd/numpy/fft.py

```diff
@@ -44,3 +71,51 @@ def check_no_repeated_axes(*args, **kwargs):
         except (TypeError, KeyError):
             # no iterable axes argument
             pass
+
+def check_even_shape(axes, shape):
+    if shape[axes[-1]] % 2 != 0:
+        raise NotImplementedError("Real FFT gradient for odd lengthed last axes is not implemented.")
+
+def find_axes(nd, n, *args, **kwargs):
```

Comment:
This logic seems complicated, and the numpy functions you're wrapping don't seem to have this logic in their code. Maybe this code would be simpler if you defined these vjps separately, but factored out the code that is shared between them into a function?